learning_journal/views.py

```diff
@@ -0,0 +1,72 @@
+from pyramid.response import Response
+from pyramid.view import view_config
+from pyramid.httpexceptions import HTTPFound
+from .form import JournalForm
+from sqlalchemy.exc import DBAPIError
+from sqlalchemy import desc
+import transaction
+import markdown
+
+from .models import (
+    DBSession,
+    Entry,
+    )
+
+
+@view_config(route_name='home', renderer='templates/list_view.jinja2')
+def list_view(request):
+    """Handle the view of our home page."""
+    return {'entries': DBSession.query(Entry).order_by(desc(Entry.created)).all()}
+
+
+@view_config(route_name='detail_view', renderer='templates/detail_view.jinja2')
+def detail_view(request):
+    """Handle the view of a single journaly entry."""
+    md = markdown.Markdown(safe_mode='replace', html_replacement_text='NO')
+    this_id = request.matchdict['this_id']
+    entry = DBSession.query(Entry).get(this_id)
+    text = md.convert(entry.text)
+    return {'entry': entry, 'text': text}
```

Comment:
What happens if you have no entry?  If `this_id` is a value that does not have a corresponding entry in the database?  What *should* happen in that case from a user's perspective?